src/GraphQL/Types/ResolveFieldContext.cs

```diff
@@ -122,11 +128,15 @@ public bool HasArgument(string argumentName)
                     er.AddLocation(FieldAst, Document);
                     er.Path = Path;
                     Errors.Add(er);
-                    return null;
+                    return default(TResult);
                 }
                 else
                 {
-                    return error(Errors);
+                    var result = error(Errors);
+                    if (result == null)
+                        return default(TResult);
+
+                    return await error(Errors);
```

Comment:
Yes, you're right. I forgot to make the change after adding the if-clause. 
I added the if-clause because 'error' returns a Task<TResult> so it should be awaited but the existing unit test returns a null (not a Task that returns null). This way both scenarios still work.